Hymns/Common/SoundCloudWebView.swift

```diff
@@ -0,0 +1,66 @@
+import SwiftUI
+import WebKit
+
+struct SoundCloudWebView: UIViewRepresentable {
+    @Binding var isLoading: Bool  //Used for activity indicator
+    var url: URL
+    var loadStatusChanged: ((Bool, Error?) -> Void)?
+
+    func makeCoordinator() -> SoundCloudWebView.Coordinator {
+        Coordinator(self)
+    }
+
+    func makeUIView(context: Context) -> WKWebView {
+        let view = WKWebView()
+        view.navigationDelegate = context.coordinator
+        view.load(URLRequest(url: url))
+        return view
+    }
+
+    func updateUIView(_ uiView: WKWebView, context: Context) {
+    }
+
+    func onLoadStatusChanged(perform: ((Bool, Error?) -> Void)?) -> some View {
+        var copy = self
+        copy.loadStatusChanged = perform
+        return copy
+    }
+
+    class Coordinator: NSObject, WKNavigationDelegate {
+        let parent: SoundCloudWebView
+
+        init(_ parent: SoundCloudWebView) {
+            self.parent = parent
+        }
+
+        func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
+            parent.loadStatusChanged?(true, nil)
+        }
+
+        func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+
+            parent.isLoading = false  //Used for activity indicator
+            webView.evaluateJavaScript("document.querySelector('.app__upsell').remove();", completionHandler: { (_, _) -> Void in
+            })
+//            webView.evaluateJavaScript("document.querySelector('.app__header').remove();", completionHandler: { (_, _) -> Void in
+//            })
```

Comment:
comment?

Reply:
I wasn't sure if we would ever want it later lol... It was a pain to get that javascript name .app__header. But I guess we can delete it?

Reply:
If it works well, we should leave it. I don't think it works well though. The "Download Soundcloud" button doesn't work for me. Does it work for you?